ibis/backends/tests/test_generic.py

```diff
@@ -1758,6 +1759,74 @@ def hash_256(col):
     backend.assert_series_equal(h1, h2)
 
 
+@pytest.mark.parametrize(
+    ("from_type", "to_type", "from_val", "expected"),
+    [
+        param("int", "float", 0, 0.0),
+        param("float", "int", 0.0, 0),
+        param("string", "int", "0", 0),
+        param("string", "float", "0", 0.0),
```

Comment:
Are you sure we aren't testing these anywhere?

Reply:
Yeah good thought. If you grep for `def test.*_cast`, you can find that there is definitely some duplication in specific backend tests. But it doesn't look like there is anything in `ibis/backends/tests/`, except for a few specific ones in `test_temporal.py`. There are a few backend-less tests in `ibis/expr/datatypes/tests/test_cast.py`. They look to be like 7 years old!

In an ideal world I would go for
1. As much as possible, we pull tests from the specific backends to be shared by all backends
2. When there is duplication, I would remove the backend-less tests, I would rather fully test for the full executed value instead of merely "does our expression system say the result is the correct type"

what do you think?

Reply:
> As much as possible, we pull tests from the specific backends to be shared by all backends

I'm on board with moving tests from the backend-specific locations to the shared locations, especially because there IS test duplication there.



> When there is duplication, I would remove the backend-less tests, I would rather fully test for the full executed value instead of merely "does our expression system say the result is the correct type"

This not so sure of.  It might be a bit duplicative when things are working, or when things are failing at the backend, but we'd want the signal that we screwed up some bit of our internals vs. wacky backend behavior.

Reply:
Sounds good.

Does this PR look acceptable as it is, and can we do this test consolidation in a separate PR?

Reply:
@gforsyth any thoughts here? would love to get this merged. Thank you!